espnet/nets/pytorch_backend/transformer/subsampling.py

```diff
@@ -98,6 +100,65 @@ def __getitem__(self, key):
         return self.out[key]
 
 
+class Conv2dSubsampling2(torch.nn.Module):
```

Comment:
Given that there are multiples Conv2Sub, I think that it could be possible to set a one single class and when calling change the kernel or stride parameters.

Reply:
Good idea. I think it's time to think about it.

Reply:
I agree. This is a very cool idea.

Reply:
@Fhrozen, is it possible for you to do it as a separate PR?
@simpleoier is preparing a follow up PR and we want to merge this PR as soon as possible.